test/WebJobs.Script.Tests.Integration/EndToEndTimeoutTests.cs

```diff
@@ -116,27 +115,23 @@ public class EndToEndTimeoutTests
             TestHelpers.ClearFunctionLogs(functionName);
             TimeSpan testTimeout = TimeSpan.FromSeconds(3);
             var traceWriter = new TestTraceWriter(TraceLevel.Verbose);
+
             using (var manager = await CreateAndStartScriptHostManager(scriptLang, functionName, testTimeout, traceWriter))
             {
                 string testData = Guid.NewGuid().ToString();
-
-                Dictionary<string, object> arguments = new Dictionary<string, object>
+                var arguments = new Dictionary<string, object>
                 {
                     { "inputData", testData },
                 };
-
-                FunctionTimeoutException ex = await Assert.ThrowsAsync<FunctionTimeoutException>(() => manager.Instance.CallAsync(functionName, arguments));
+                var timeoutException = await Assert.ThrowsAsync<FunctionTimeoutException>(() => manager.Instance.CallAsync(functionName, arguments));
 
                 await TestHelpers.Await(() =>
                 {
-                    // make sure logging from within the function worked
-                    // TODO: This doesn't appear to work for Powershell in AppVeyor. Need to investigate.
-                    // bool hasTestData = inProgressLogs.Any(l => l.Contains(testData));
-                    var expectedMessage = $"Timeout value of {testTimeout} was exceeded by function: Functions.{functionName}";
-                    var traces = string.Join(Environment.NewLine, traceWriter.GetTraces());
-                    return traces.Contains(expectedMessage);
+                    // wait for the timeout exception to be logged
+                    return traceWriter.GetTraces().Any(p => p.Message == $"Microsoft.Azure.WebJobs.Host: Timeout value of {testTimeout} was exceeded by function: Functions.{functionName}.");
                 });
 
+                // verify that our mock handler received the timeout exception
```

Comment:
This .Single() is what was throwing, because there were multiple timeouts in rare cases